### ISSUE TITLE
[BUG] Some or every QT apps will not open

### ISSUE THREAD

**Issue by pinarruiz** [danth/stylix#487]
# Context
Hi,

Until now i have been using [bluskript/stylix](https://github.com/bluskript/stylix) from #142 as non of the QT apps had the stylix theme applied and this fork worked.

I saw that recently the https://github.com/danth/stylix/commit/6bbae4f85b891df2e6e48b649919420434088507 commit should fix the issue, but instead i am getting this error in what seems every QT app that I open:

```console
QSystemTrayIcon::setVisible: No Icon set
```

Particularly Nextcloud and VLC, I dont know/think I have any others, but shows the error on both.

My stylix config:
```nix
{ pkgs, ... }: {
  stylix = {
    enable = true;
    polarity = "dark";
    image = pkgs.fetchurl {
      url = "https://...";
      sha256 = "...";
    };
  };
}
```

## Versions

`stylix`: https://github.com/danth/stylix/commit/0ef70039a6435446472182c8f8106947abfc523d
`nixpkgs`: [stable-24-05 (`63d37cc`)](https://github.com/NixOS/nixpkgs/tree/63d37ccd2d178d54e7fb691d7ec76000740ea24a)
`Home Manager`: [release-24.05 (`e1391fb`)](https://github.com/nix-community/home-manager/commit/e1391fb22e18a36f57e6999c7a9f966dc80ac073)

## Additional Context

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.41, NixOS, 24.05 (Uakari), 24.05.20240721.63d37cc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/lzhqf5y9dcgfkg5f7f96jjc814pgrsqy-source`
```

Nextcloud client in Home Manager:
```nix
{ lib, ... }: {
  services.nextcloud-client = {
    enable = lib.mkDefault true;
    startInBackground = lib.mkDefault true;
  };
}
```

VLC was added as a system package:
```nix
{ pkgs, ... }: {
  environment.systemPackages = with pkgs; [
    ...
    vlc
    ...
  ];
}
```

Related: #367 

If any extra context is required please let me know.

Thanks for the help in advance :smile:.

**Comment by trueNAHO** [comment]
@Jackaed, do you maybe know how to resolve this?

**Comment by Jackaed** [comment]
No idea while I'm away from my computer, I think I tested VLC on my system so I'm surprised it isn't working. I'll have a look.

**Comment by Jackaed** [comment]
I can't reproduce this on unstable. Could you try a "more normal" Qt app (just for my information), e.g. dolphin or pcmanfm-qt?

**Comment by danth** [comment]
Does VLC use a system tray icon? (That's the ones that appear in the taskbar usually to show that an app is running in the background.) I know Nextcloud does and that seems to be what the error is hinting towards. Perhaps we have accidentally overwritten their icons somehow by installing the Breeze icons package?

**Comment by pinarruiz** [comment]
> I can't reproduce this on unstable. Could you try a "more normal" Qt app (just for my information), e.g. dolphin or pcmanfm-qt?

Hi, @Jackaed 

Will try later in the day, as I cannot do it now, and will give some feedback, the vlc version i am using is the one in [stable-24-05 (63d37cc)](https://github.com/NixOS/nixpkgs/tree/63d37ccd2d178d54e7fb691d7ec76000740ea24a), but i have my config ready for allowing some unstable packages, I will let you know how that goes.

**Comment by pinarruiz** [comment]
> Does VLC use a system tray icon? (That's the ones that appear in the taskbar usually to show that an app is running in the background.) I know Nextcloud does and that seems to be what the error is hinting towards. Perhaps we have accidentally overwritten their icons somehow by installing the Breeze icons package?

Hi, @danth 

I don't know if VLC has a tray icon, i cant remember ever seeing one, that said, https://github.com/danth/stylix/pull/142 seems to work, and i guess maybe the theme is the culprit as @Jackaed is suggesting (https://github.com/danth/stylix/issues/487#issuecomment-2248971438)

**Comment by pinarruiz** [comment]
Hi @Jackaed 

Just checked with unstable vlc (versions seem to be the same) and i got the same result. The app appears not to launch and it gives the same error message:

```console
QSystemTrayIcon::setVisible: No Icon set
```

**Comment by pinarruiz** [comment]
I am not sure #490 should close this issue, it will render it useless thats true, but disabling a feature that's not working should not be a reason to close an issue at least in my opinion, what do you think @trueNAHO ?

**Comment by trueNAHO** [comment]
> I am not sure #490 should close this issue, it will render it useless thats true, but disabling a feature that's not working should not be a reason to close an issue at least in my opinion, what do you think @trueNAHO ?

Closing issues by reverting the commits introducing them is somewhat controversial.

IMHO, issues related to #490 should be closed to distinguish existing and non-existing problems. However, the issues closed by #490 will serve as references and considerations for #489.

**Comment by pinarruiz** [comment]
Makes sense to me, adding it to the related section at #489 looks good, so all good from my part closing the issue with #490 as its being referenced on the mentioned tracking issue.
